### PR TITLE
treewide: Remove some bashism

### PR DESCRIPTION
**Pull request by SvenRoederer**
This removes some non POSIX extensions from the shell-scripts, as already done in d6ac8ca76c04ed39d7dd5de73d3b921819934186. It targets "[[" and "==".

### CODE REVIEW

File: package/base-files/files/lib/functions/caldata.sh [hunk 1]
```diff
@@ -69,7 +69,7 @@ caldata_valid() {
 	local expected="$1"
 
 	magic=$(hexdump -v -n 2 -e '1/1 "%02x"' /lib/firmware/$FIRMWARE)
-	[[ "$magic" == "$expected" ]]
+	[ "$magic" == "$expected" ]
```
Comment (adrianschmutzler) [inline package/base-files/files/lib/functions/caldata.sh]: You can use a single equal sign as well.

**Comment by adrianschmutzler** [comment]
For commit messages like "```[[``` is a bash-extension for test the ash-implementation is in fully compatible." I think it makes more sense to just put the "[[" in standard quotes.

**Comment by neheb** [comment]
If bashisms are to be removed, it makes sense to change the top shebang to point to /bin/sh as dash is faster than bash.

**Comment by adrianschmutzler** [comment]
@SvenRoederer Regarding "-a":
https://patchwork.ozlabs.org/patch/1216221/

**Comment by SvenRoederer** [comment]
@neheb  I added the shebang to the files I edited. The scripts I edited are only running on the target-board, so /bin/sh links to busyboy-ash, right?

@adrianschmutzler Regarding "-a": see 01c84b4

**Comment by adrianschmutzler** [comment]
@SvenRoederer I think it will be more efficient if you already provide properly formatted patches in the first place. That will allow me to pick what's ready (or looks correct) right away. Currently, I would have to look at rather "trivial" patches several times, but I do not think this will benefit so much from having it pre-reviewed.
However, feel free to stick to your current plan if you prefer.

**Comment by adrianschmutzler** [comment]
Ping.

**Comment by SvenRoederer** [comment]
@adrianschmutzler commits db23f92 , 38bebf5 , aeb858d , 0cc2d7c can be cherry-picked.

The commits related to igmpproxy should be reviewed.

File: package/base-files/files/lib/functions/caldata.sh [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#!/bin/sh
```
Comment (adrianschmutzler) [inline package/base-files/files/lib/functions/caldata.sh]: These are library files, which are non-executable, and thus should not contain a shebang.

**Comment by adrianschmutzler** [comment]
Note that the author of your patches (From:) does not match your SOB. I corrected that manually for those patches already merged, but please take care of future patches yourself.

**Comment by SvenRoederer** [comment]
Adding the shebang was based on this comment https://github.com/openwrt/openwrt/pull/2647#issuecomment-569171959.

I changed my github settings, which hopefully improves the "From:" header.

**Comment by SvenRoederer** [comment]
@adrianschmutzler I just got a ping for this PR and reworked commits according to your comments. I also tested the change of the 2 commit to give the same results as the original.

**Comment by adrianschmutzler** [comment]
Hmm, just had another look and I still don't like it. The first commit tells to change brackets, but also changes &&/-a and -z/-n, only to have it changed again in the second commit.

Please separate these changes properly. Despite, be aware that I'm treating this low-priority as it's cosmetic.

Thanks for your work, however.

**Comment by adrianschmutzler** [comment]
I've squashed the two patches together with a slightly modified message and will merge them later today.

**Comment by SvenRoederer** [comment]
Thanks for taking care.... I had it on my list with a list with a low priority ...

**Pull request closed without merging**